### PR TITLE
Handle zero-extent dataset creation and dataset write

### LINKED ISSUE #135: Test cases with 0-sized reads & writes
To verify the desired parallel behaviour, we should add tests where multiple participants from an MPI-communicator contribute 0-sized chunks to a collective IO operation.

A minimal example would consist of 3 ranks with multiple tested combinations of empty/non-empty chunks.

### PR DESCRIPTION
**Pull request by anokfireball**
Adds tests for #135.

### CODE REVIEW

File: test/CoreTest.cpp [hunk 1]
```diff
@@ -578,3 +578,18 @@ TEST_CASE( "use_count_test", "[core]" )
     REQUIRE(static_cast< Parameter< Operation::WRITE_DATASET >* >(pprc.m_chunks->front().parameter.get())->data.use_count() == 1);
 #endif
 }
+
+TEST_CASE( "zero_extent_component", "[core]" )
+{
+    Series o = Series("./new_openpmd_output", AccessType::CREATE);
+
+    auto E_x = o.iterations[1].meshes["E"]["x"];
+    E_x.setComment("Datasets with zero extent in any dimension are not allowed.");
```
Comment (anokfireball) [inline test/CoreTest.cpp:628]: ~We should also enforce that the extent is at least 1D.~
Done.

**Comment by ax3l** [comment]
@C0nsultant can you rebase this PR pls? :)

File: test/ParallelIOTest.cpp [hunk 2]
```diff
@@ -140,6 +140,60 @@ TEST_CASE( "hdf5_write_test", "[parallel][hdf5]" )
...
+    Series o = Series("../samples/parallel_write_zero_extent.h5", AccessType::CREATE, MPI_COMM_WORLD);
+
+    ParticleSpecies& e = o.iterations[1].particles["e"];
+
+    /* every rank n writes n consecutive cells, increasing values
+     * rank 0 does a zero-extent write
+     * two ranks will result in {1}
+     * three ranks will result in {1, 2, 3}
+     * four ranks will result in {1, 2, 3, 4, 5, 6} */
+    uint64_t num_cells = ((size-1)*(size-1) + (size-1))/2; /* (n² + n) / 2 */
```
Comment (ax3l) [inline test/ParallelIOTest.cpp]: hm, maybe use `n^2` here to stay in full-ascii source? had troubles with some GCC versions in the past with non-ascii chars in source files.

File: test/ParallelIOTest.cpp [hunk 3]
```diff
@@ -140,6 +140,60 @@ TEST_CASE( "hdf5_write_test", "[parallel][hdf5]" )
...
+
+    for( uint64_t i = 0; i < rank; ++i )
+    {
+        position_local.get()[i] = position_global[offset + i];
+        positionOffset_local.get()[i] = positionOffset_global[offset + i];
+    }
+    e["position"]["x"].storeChunk({offset}, {rank}, position_local);
+    e["positionOffset"]["x"].storeChunk({offset}, {rank}, positionOffset_local);
+
+    //TODO read back, verify
```
Comment (ax3l) [inline test/ParallelIOTest.cpp:195]: forgotten TODO?

File: test/ParallelIOTest.cpp [hunk 4]
```diff
@@ -182,6 +236,60 @@ TEST_CASE( "adios_write_test", "[parallel][adios]" )
...
+    Series o = Series("../samples/parallel_write_zero_extent.bp", AccessType::CREATE, MPI_COMM_WORLD);
+
+    ParticleSpecies& e = o.iterations[1].particles["e"];
+
+    /* every rank n writes n consecutive cells, increasing values
+     * rank 0 does a zero-extent write
+     * two ranks will result in {1}
+     * three ranks will result in {1, 2, 3}
+     * four ranks will result in {1, 2, 3, 4, 5, 6} */
+    uint64_t num_cells = ((size-1)*(size-1) + (size-1))/2; /* (n² + n) / 2 */
```
Comment (ax3l) [inline test/ParallelIOTest.cpp]: ASCII :)

File: test/ParallelIOTest.cpp [hunk 5]
```diff
@@ -182,6 +236,60 @@ TEST_CASE( "adios_write_test", "[parallel][adios]" )
...
+
+    for( uint64_t i = 0; i < rank; ++i )
+    {
+        position_local.get()[i] = position_global[offset + i];
+        positionOffset_local.get()[i] = positionOffset_global[offset + i];
+    }
+    e["position"]["x"].storeChunk({offset}, {rank}, position_local);
+    e["positionOffset"]["x"].storeChunk({offset}, {rank}, positionOffset_local);
+
+    //TODO read back, verify
```
Comment (ax3l) [inline test/ParallelIOTest.cpp:290]: forgotten TODO or for later?

**Review by ax3l — APPROVED** [review]

File: test/ParallelIOTest.cpp (hunk 2 above)
Comment (anokfireball) [reply test/ParallelIOTest.cpp]: CLion was *smart* enough to make it superscript.

File: test/ParallelIOTest.cpp (hunk 3 above)
Comment (anokfireball) [reply test/ParallelIOTest.cpp:195]: I manually checked the contents of the files (with ```h5dump``` and ```bpdump```), but we should strive to have this verification in-code.

**Pull request merged**